In [2]:
%matplotlib inline
import pandas as pd
import numpy as np

In [3]:
cast = pd.read_csv("cast.csv")
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4   NaN
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

In [4]:
titles = pd.read_csv("titles.csv")
titles.head()

title  year
0           The Rising Son  1990
1  The Thousand Plane Raid  1969
2         Crucea de piatra  1993
3                  Country  2000
4               Gaiking II  2011

### How many movies are listed in the titles dataframe?

In [5]:
num_movies = len(titles["title"].unique())
print("Number of movies are listed in the titles dataframe is : ",num_movies)

Number of movies are listed in the titles dataframe is :  210938


**2,10,938 movies are listed in this titles dataframe**

### What are the earliest two films listed in the titles dataframe?

In [6]:
earliest_movies = titles.sort_values(by="year")["title"][:2]
print(" Earliest two films listed in the titles dataframe are : ",earliest_movies)

 Earliest two films listed in the titles dataframe are :  176477            Miss Jerry
213694    The Startled Lover
Name: title, dtype: object


**Earliest two films in the titles dataframes are**
* **Miss Jerry**
* **TheStartled Lover**

### How many movies have the title "Hamlet"?

In [7]:
movie_hamlet = len(titles[titles.title == "Hamlet"])
print("The number of Movies have the title 'Hamlet' are : ",movie_hamlet)


The number of Movies have the title 'Hamlet' are :  18


**18 Movies have the title of "Hamlet"**

### How many movies are titled "North by Northwest"?

In [8]:
movie_NBN = len(titles[titles.title == "North by Northwest"])
print("The number of Movies have the title 'North by Northwest' are : ",movie_NBN)

The number of Movies have the title 'North by Northwest' are :  1


**1 movie have the title of "North by Northwest"**

### When was the first movie titled "Hamlet" made?

In [9]:
first_made = titles[titles.title == "Hamlet"].year.min()
print("The Movie titled 'Hamlet' made in the year of : ",first_made)

The Movie titled 'Hamlet' made in the year of :  1910


**In 1910 the movie titled "Hamlet" was made**

### List all of the "Treasure Island" movies from earliest to most recent.

In [10]:
Treasure_island = titles[titles.title == "Treasure Island"].sort_values(by='year')
Treasure_island

title  year
204540  Treasure Island  1918
50896   Treasure Island  1920
189691  Treasure Island  1934
96259   Treasure Island  1950
59186   Treasure Island  1972
110576  Treasure Island  1973
203909  Treasure Island  1985
178072  Treasure Island  1999

### How many movies were made in the year 1950?

In [11]:
yr_1950 = len(titles[titles.year == 1950])
print("Number of movies were made in the year of 1950 : ",yr_1950)

Number of movies were made in the year of 1950 :  1093


**1,093 movies are made in the year of 1950**

### How many movies were made in the year 1960?

In [12]:
yr_1960 = len(titles[titles.year == 1960])
print("Number of movies were made in the year of 1950 : ",yr_1960)

Number of movies were made in the year of 1950 :  1496


**1,496 movies are made in the year of 1950**

### How many movies were made from 1950 through 1959?

In [13]:
from_50_to_59 = len(titles[(titles.year >= 1950) & (titles.year <= 1959)])
print("Number of movies were made from 1950 to 1959 are : ",from_50_to_59 )

Number of movies were made from 1950 to 1959 are :  12656


**12,656 Movies are made from 1950 to 1959**

### In what years has a movie titled "Batman" been released?

In [14]:
Batman_movie = titles[titles.title == "Batman"]
print("A movie titled 'Batman' has been released in the year of : ",Batman_movie)

A movie titled 'Batman' has been released in the year of :          title  year
55577  Batman  1989
74047  Batman  1943


**In the year of 1989 and 1943 movie titled "Batman" was released**

### How many roles were there in the movie "Inception"?

In [15]:
role_inception = len(cast[cast.title == "Inception"])
print("Number of roles were there in the movie 'Inception' are : ",role_inception)

Number of roles were there in the movie 'Inception' are :  77


**77 roles are there in the movie "Inception"**

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

not_n_value = len(cast[(cast.title == "Inception") & (cast.n.isnull())])
print(not_n_value)

**26 roles are in the movie "Inception" are not ranked by an "n" value**

### But how many roles in the movie "Inception" did receive an "n" value?

In [18]:
n_value = len(cast[(cast.title == "Inception") & (cast.n.notnull())])
print(n_value)

51


**51 roles are in the movie "Inception" are ranked by an "n" value**

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [19]:
cast_of_NBN = cast[(cast.title == "North by Northwest") & (cast.n.notnull())].sort_values(by='n')
print(cast_of_NBN)

                      title  year                  name     type  \
840374   North by Northwest  1959            Cary Grant    actor   
3361396  North by Northwest  1959       Eva Marie Saint  actress   
1404704  North by Northwest  1959           James Mason    actor   
3024085  North by Northwest  1959   Jessie Royce Landis  actress   
343701   North by Northwest  1959        Leo G. Carroll    actor   
2923056  North by Northwest  1959  Josephine Hutchinson  actress   
1634552  North by Northwest  1959           Philip Ober    actor   
1229083  North by Northwest  1959         Martin Landau    actor   
2357939  North by Northwest  1959         Adam Williams    actor   
1746383  North by Northwest  1959          Edward Platt    actor   
642085   North by Northwest  1959     Robert Ellenstein    actor   
2212083  North by Northwest  1959          Les Tremayne    actor   
447893   North by Northwest  1959       Philip Coolidge    actor   
1454363  North by Northwest  1959         Patric

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [20]:
cast_of_Sleuth = cast[(cast.title == "Sleuth") & (cast.year == 1972) & (cast.n.notnull())].sort_values(by='n')
print(cast_of_Sleuth)

          title  year                name     type  \
1644328  Sleuth  1972    Laurence Olivier    actor   
314691   Sleuth  1972       Michael Caine    actor   
360365   Sleuth  1972      Alec Cawthorne    actor   
1412647  Sleuth  1972  John (II) Matthews    actor   
2619278  Sleuth  1972  Eve (III) Channing  actress   
1396572  Sleuth  1972        Teddy Martin    actor   

                          character    n  
1644328                 Andrew Wyke  1.0  
314691                  Milo Tindle  2.0  
360365            Inspector Doppler  3.0  
1412647  Detective Sergeant Tarrant  4.0  
2619278             Marguerite Wyke  5.0  
1396572      Police Constable Higgs  6.0  


### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [21]:
cast_of_sleuth_2007 = cast[(cast.title == "Sleuth") & (cast.year == 2007) & (cast.n.notnull())].sort_values(by='n')
print(cast_of_sleuth_2007 )

          title  year           name   type    character    n
314692   Sleuth  2007  Michael Caine  actor       Andrew  1.0
1246855  Sleuth  2007       Jude Law  actor         Milo  2.0
1741124  Sleuth  2007  Harold Pinter  actor  Man on T.V.  3.0


### How many roles were credited in the silent 1921 version of Hamlet?

In [22]:
roles_of_hamlet = len(cast[(cast.title == "Hamlet") & (cast.year == 1921)])
print(roles_of_hamlet)

9


**9 roles were credited in the silent 1921version of "Hamlet"**

### How many roles were credited in Branagh’s 1996 Hamlet?

In [24]:
roles_of_hamlet_1996 = len(cast[(cast.title == "Hamlet") & (cast.year == 1996)])
print(roles_of_hamlet_1996)

54


**54 roles were credited in Branagh's 1996 hamlet**

### How many "Hamlet" roles have been listed in all film credits through history?

In [26]:
credits_hamlet = len(cast[cast.character == "Hamlet"])
print(credits_hamlet)

91


**91 Hamlet roles have been listed in all film credits through history**

### How many people have played an "Ophelia"?

In [27]:
ophelia = len(cast[cast.character == "Ophelia"])
print(ophelia)

111


**111 people have played an "Ophelia"**

### How many people have played a role called "The Dude"?

In [28]:
The_Dude = len(cast[cast.character == "The Dude"])
print(The_Dude)

18


**18 people have played a role called "The Dude"**

### How many people have played a role called "The Stranger"?

In [29]:
The_Stranger = len(cast[cast.character == "The Stranger"])
print(The_Stranger)

208


**208 people have played a role called "The Stranger"**

### How many roles has Sidney Poitier played throughout his career?

In [30]:
sidney_poitier = len(cast[cast.name == "Sidney Poitier"])
print(sidney_poitier)

43


**43 roles has  been played by Sidney Poitier throughout his career**

### How many roles has Judi Dench played?

In [31]:
Judi_Dench = len(cast[cast.name == "Judi Dench"])
print(Judi_Dench)

55


**55 roles has been played by Judi Dench**

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [33]:
cast[(cast.name == 'Cary Grant') & 
     (cast.year // 10 == 194) &
     (cast.n == 2)].sort_values(by='year')

title  year        name   type    character    n
840371  My Favorite Wife  1940  Cary Grant  actor         Nick  2.0
840381    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [34]:
cast[(cast.name == 'Cary Grant') & 
     (cast.year // 10 == 194) &
     (cast.n == 1)].sort_values(by='year')

title  year        name   type  \
840397               The Howards of Virginia  1940  Cary Grant  actor   
840353                       His Girl Friday  1940  Cary Grant  actor   
840399                The Philadelphia Story  1940  Cary Grant  actor   
840386                             Suspicion  1941  Cary Grant  actor   
840401                  The Talk of the Town  1942  Cary Grant  actor   
840377                 Once Upon a Honeymoon  1942  Cary Grant  actor   
840344                     Destination Tokyo  1943  Cary Grant  actor   
840369                             Mr. Lucky  1943  Cary Grant  actor   
840370                             Mr. Lucky  1943  Cary Grant  actor   
840378                      Once Upon a Time  1944  Cary Grant  actor   
840336                  Arsenic and Old Lace  1944  Cary Grant  actor   
840373             None But the Lonely Heart  1944  Cary Grant  actor   
840372                         Night and Day  1946  Cary Grant  actor   
840375                             Notorious  1946  Cary Grant  actor   
840393      The Bachelor and the Bobby-Soxer  1947  Cary Grant  actor   
840394                     The Bishop's Wife  1947  Cary Grant  actor   
840368  Mr. Blandings Builds His Dream House  1948  Cary Grant  actor   
840348          Every Girl Should Be Married  1948  Cary Grant  actor   
840357                I Was a Male War Bride  1949  Cary Grant  actor   

                    character    n  
840397            Matt Howard  1.0  
840353           Walter Burns  1.0  
840399     C. K. Dexter Haven  1.0  
840386                Johnnie  1.0  
840401           Leopold Dilg  1.0  
840377  Patrick 'Pat' O'Toole  1.0  
840344          Capt. Cassidy  1.0  
840369              Joe Adams  1.0  
840370        Joe Bascopolous  1.0  
840378            Jerry Flynn  1.0  
840336      Mortimer Brewster  1.0  
840373             Ernie Mott  1.0  
840372            Cole Porter  1.0  
840375                 Devlin  1.0  
840393            Dick Nugent  1.0  
840394                 Dudley  1.0  
840368          Jim Blandings  1.0  
840348      Dr. Madison Brown  1.0  
840357  Captain Henri Rochard  1.0

### How many roles were available for actors in the 1950s?

In [36]:
roles_in_1950 = len(cast[(cast.year // 10 == 195) &
         (cast.type == "actor")])
print(roles_in_1950)

154727


**1,54,727 roles were avalible for actors in 1950s**

### How many roles were available for actresses in the 1950s?

In [37]:
roles_for_actress = len(cast[(cast.year // 10 == 195) &
         (cast.type == "actress")])
print(roles_for_actress)

56259


**56,259 roles were available for actresses in 1950s**

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [38]:
leading_roles_1980 = len(cast[(cast.year <= 1980) &
                              (cast.n == 1)])
print(leading_roles_1980)

63924


**63,924 leading roles(n=1) were availbale from the beginning of film history through 1980** 

### How many non-leading roles were available through from the beginning of film history through 1980?

In [39]:
non_leading_roles_80s = len(cast[(cast.year <= 1980) &
                                 (cast.n != 1)])
print(non_leading_roles_80s)

1094780


**10,94,780 non-leading roles were available through from the beginning of film history through 1980**

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [40]:
minor_role = len(cast[(cast.year <= 1980) &
                      (cast.n.isnull())])
print(minor_role)

438086


**4,38,086 roles through 1980 were minor enough that they did not warrant a numeric "n"**